examples/features/src/main/java/com/github/kagkarlsson/examples/kotlin/KotlinSerializer.kt

```diff
@@ -10,11 +10,23 @@ class KotlinSerializer : Serializer {
     val CHARSET = StandardCharsets.UTF_8
 
     override fun serialize(data: Any): ByteArray {
+        if (data == null) {
+            return ByteArray(0)
+        }
         val serializer = serializer(data.javaClass)
         return Json.encodeToString(serializer, data).toByteArray(CHARSET);
     }
 
     override fun <T : Any?> deserialize(clazz: Class<T>, serializedData: ByteArray): T {
+        if (serializedData == null || clazz == Void::class.java) {
+            return null
+        }
+
+        // If we do not know the class (e.g. java.lang.Object), decode as generic JSON
+        if (clazz == Any::class.java) {
+            return Json.decodeFromString(JsonElement.serializer(), serializedData.decodeToString()) as T
```

Comment:
When do we not know the class?  🤔

Reply:
I'm not quite sure why or how it happened, but it did 😅

Reply:
That is, we know the class, but the class is java.lang.Object

Reply:
Clarified the comment.